presto-main/src/main/java/com/facebook/presto/operator/ExchangeClientFactory.java

```diff
@@ -66,7 +67,11 @@ public ExchangeClientFactory(
         this.concurrentRequestMultiplier = concurrentRequestMultiplier;
         this.minErrorDuration = requireNonNull(minErrorDuration, "minErrorDuration is null");
         this.httpClient = requireNonNull(httpClient, "httpClient is null");
-        this.maxResponseSize = requireNonNull(maxResponseSize, "maxResponseSize is null");
+
+        requireNonNull(maxResponseSize, "maxResponseSize is null");
+        long maxResponseSizeBytes = Math.min(httpClient.getMaxContentLength(), maxResponseSize.toBytes()) / 2;
```

Comment:
So we're going with 50%?

Please add a comment and a TODO explaining that this is a temporary fix that needs to be revisited.